wordpress-mobile/WordPress-iOS
# Show SignIn as root VC.

This PR changes our logic to show the sign in flow as our root VCs.

This also fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/15399 (thanks @jaclync for pointing this out!)

## The short explanation

This is an example of what this PR resolves:

![startupGlitch](https://user-images.githubusercontent.com/1836005/105498936-8cfedb00-5c9f-11eb-862a-0d64213f0687.gif)

## The long explanation

While working in [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/15281) I noticed the App's UI was sometimes showing up for a moment before the login UI was visible.  Something similar to what @jaclync reported [here](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/564#pullrequestreview-573716784).  My fix is actually (kind of) similar to what Jaclyn is pushed [here](https://github.com/woocommerce/woocommerce-ios/issues/3429).

Additionally, [in this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/15281), I'm having a bit of trouble with some view controllers that are being updated behind the login UI while there's no user.  This is a scenario that we've discussed in the past with @frosty.

This PR is probably not enough to kill the full UI while logged out, since WPiOS uses a shared `WPTabBarController` which is never really deleted.  But it's a very strong (IMHO) first step towards simplifying out UI updates while there's no user signed in.

In short this aligns the lifecycle of the UI with the authentication state of the user.

## Animation Changes:

While I didn't intend to change anything animation-wise, I found that the previous transitions we used for swapping between full-screen sign in and the regular App UI require some custom code with the changes I'm introducing in this PR.  We can implement them as they were but they'll require some extra coding.

In weighting this, I also noticed the new animation looks kinda good IMHO - but I'm not sure if it's just me so I'm posting a demo to get direction on it from our designers.

@mbshakti can I ask you to let me know if it'd be acceptable to use this transition, or if you want me to roll back to the regular slide-down animation we previously had?

![newSignInTransition](https://user-images.githubusercontent.com/1836005/105443494-9ad64100-5c4a-11eb-8455-741d143fbb74.gif)

## To test:

1. Test logging in from scratch:
   - [ ] Try the standard case where you just log in and dismiss the login epilogue directly.
   - [ ] Try the variant where in the login epilogue you add another site by address.
   - [ ] Try the variant where the account you use to log in has no sites - this should show a different epilogue.
2. Test signing up:
   - [ ] Include the standard case in which you just sign up and dismiss the epilogue.
   - [ ] Include the variant in which you create a site in the signup epilogue.
   - [ ] Include the variant in which you add site by address in the signup epilogue.
3. Test adding a self hosted site in the "My Sites" tab, when you're already logged in.
   - [ ] Test you're able to add a self hosted site just fine.
   - [ ] When using an iPad the login UI should be shown "inline" instead of full screen.
4. Test disconnecting the WPiOS session from your WP.com Security Settings to see how the App reacts to it.
   - [ ] Test you're able to reconnect your session just fine.
   - [ ] When using an iPad the login UI should be shown "inline" instead of full screen.

Make sure you test everything in both iPad and iPhone.

## PR submission checklist:
